contracts/orbital-core/src/tests/unit_tests.rs

```diff
@@ -146,21 +169,53 @@ fn test_register_orbital_domain_happy() {
         )
         .unwrap();
 
-    let domain: OrbitalDomainConfig = suite
+    suite
+        .app
+        .execute_contract(
+            suite.owner,
+            suite.orbital.clone(),
+            &ExecuteMsg::RegisterNewDomain {
+                domain: "domain_ica".to_string(),
+                account_type: AccountConfigType::InterchainAccount {
+                    connection_id: "connection-id".to_string(),
+                    channel_id: "channel-id".to_string(),
+                    timeout: Uint64::one(),
+                },
+            },
+            &[],
+        )
+        .unwrap();
+
+    let polytone_domain: OrbitalDomainConfig = suite
         .app
         .wrap()
         .query_wasm_smart(
             suite.orbital.clone(),
             &QueryMsg::OrbitalDomain {
-                domain: "domain".to_string(),
+                domain: "domain_polytone".to_string(),
+            },
+        )
+        .unwrap();
+    let ica_domain: OrbitalDomainConfig = suite
+        .app
+        .wrap()
+        .query_wasm_smart(
+            suite.orbital.clone(),
+            &QueryMsg::OrbitalDomain {
+                domain: "domain_ica".to_string(),
             },
         )
         .unwrap();
 
-    match domain {
-        OrbitalDomainConfig::Polytone { note, timeout } => {
-            assert_eq!(note, suite.note);
-            assert_eq!(timeout, Uint64::one());
-        }
-    }
+    assert!(matches!(
+        polytone_domain,
+        OrbitalDomainConfig::Polytone { note, timeout }
+        if note == suite.note && timeout == Uint64::one()
+    ));
+
+    assert!(matches!(
+        ica_domain,
+        OrbitalDomainConfig::ICA { connection_id, channel_id, timeout }
+        if connection_id == "connection-id" && channel_id == "channel-id" && timeout == Uint64::one()
+    ));
```

Comment:
Do we need the matches? can't you do `assert!(polytone_domain == OrbitalDomainConfig::Polytone { note, timeout })` and derive partialEq on it?

Reply:
yeah that's true, partialEq is already there from cw_serde annotation. applied!